src/coreclr/src/jit/gentree.cpp

```diff
@@ -5483,6 +5483,18 @@ bool GenTree::OperMayThrow(Compiler* comp)
         case GT_ARR_ELEM:
             return comp->fgAddrCouldBeNull(this->AsArrElem()->gtArrObj);
 
+        case GT_FIELD:
+        {
+            GenTree* fldObj = this->AsField()->gtFldObj;
+
+            if (fldObj != nullptr)
+            {
+                return comp->fgAddrCouldBeNull(fldObj);
```

Comment:
This was presumably needed because this is now called before morph has removed these?

Reply:
Yes.